sdks/java/harness/src/main/java/org/apache/beam/fn/harness/control/ProcessBundleHandler.java

```diff
@@ -255,6 +258,19 @@ private void createRunnerAndConsumersForPTransformRecursively(
       // Create a BeamFnStateClient
       for (Map.Entry<String, RunnerApi.PTransform> entry :
           bundleDescriptor.getTransformsMap().entrySet()) {
+
+        if (RemoteGrpcPortWrite.isSinkPtransform(entry.getValue())) {
+          RemoteGrpcPortWrite rgpw = RemoteGrpcPortWrite.fromPTransform(entry.getValue());
+          if (apiServiceDescriptor == null) {
+            apiServiceDescriptor = rgpw.getPort().getApiServiceDescriptor();
+          }
+
+          if (!apiServiceDescriptor.equals(rgpw.getPort().getApiServiceDescriptor())) {
+            throw new Exception(
```

Comment:
use, UnsupportedOperationException containing a TODO comment about supporting multiple ApiServiceDescriptors during reading in the future.

Reply:
RM'd

Reply:
RM'd